Textfield/Textfield.jsx

```diff
@@ -64,33 +68,58 @@ class TextfieldInput extends MaterialComponent {
       }
     );
   }
+
   componentWillUnmount() {
     this.MDComponent && this.MDComponent.destroy && this.MDComponent.destroy();
   }
-  materialDom(allprops) {
+  
+  _onFocus() {
+    this.setState({
+      isFocused: true
+    });
+  }
+
+  _onBlur() {
+    this.setState({
+      isFocused: false
+    });
+  }
+
+  materialDom(allprops, { showFloatingLabel, isFocused }) {
     let { className, ...props } = allprops;
+    let labelClassName = "mdc-textfield__label";
+
+    const upgrade = props.value || isFocused;
 
-    if (props.value && this.state.showFloatingLabel) {
+    if (showFloatingLabel) {
       className = [className, "mdc-textfield--upgraded"].join(" ");
     }
 
+    if (upgrade) {
+      labelClassName = [labelClassName, "mdc-textfield__label--float-above"].join(" ");
+    }
+
     return (
       <div className={className} ref={control => (this.control = control)}>
         {props.multiline
           ? <textarea
               className="mdc-textfield__input"
+              onFocus={this._onFocus}
+              onBlur={this._onBlur}
               {...props}
             />
           : <input
               type={props.type || "text"}
               className="mdc-textfield__input"
+              onFocus={this._onFocus}
+              onBlur={this._onBlur}
               {...props}
             />}
         {props.label &&
           this.state.showFloatingLabel &&
           <Label
             for={props.id}
-            className={props.value && "mdc-textfield__label--float-above"}
```

Comment:
`props.value` check is needed cuz it floats above only when some value is there. float on focus is auto handled by `mdc-web`